### ISSUE TITLE
Running yield generator in the long term

### ISSUE THREAD

**Issue by bulldozer2003** [JoinMarket-Org/joinmarket#536]
Are there any recommendations for handling a yield generator in the long term? I'm talking about when the watch-only wallet in bitcoind is approaching 10,000 addresses. This must put some load on bitcoind, although I can't quantify how much.

Obviously, consolidating the wallet into a a few addresses in a new wallet is an option, but would reduce the anonymity of every CJ your yieldgen was part of. Or, if I've read the documentation correctly, consolidating each mix level into the same level of a new wallet would be okay?

**Comment by chris-belcher** [comment]
Are you saying you have 10,000 UTXOs? Or 10,000 spent addresses?

If you have that many UTXOs, set `merge_algorithm = greedy` in joinmarket.cfg and then set it back to `merge_algorithm = default` when you get a more reasonable size. We've got a PR in the works #414 which would do all this automatically.

If it's 10,000 addresses it probably wont be a problem. Bitcoin exchanges have far more than that.

**Comment by bulldozer2003** [comment]
10k addresses fortunately/unfortunately, you probably need a few hundred BTC to have that many unspent UTXOs I imagine. :godmode:

Though, I'm not sure if it doesn't affect bitcoind, I once piled a few hundred thousand private keys into bitcoind 0.8 or 0.9 and it was slowing to a crawl.

**Comment by bulldozer2003** [comment]
If you did want to move to a fresh wallet, for any reason, what would the somewhat privacy preserving method be?

1. Dumping the entire wallet into m/0/0/0/000 of the new wallet (probably not...)
1. Moving by mix level in aggregate (i.e. m/0/0..4/ => m/0/0..4/0/000)
2. Moving by mix level and change in aggregate (m/0/0..4/0..1/ => m/0/0..4/0..1/000)
3. Moving each UTXO to the new wallet individually (m/0/0..4/0..1/n..n+x => m/0/0..4/0..1/0..x)

**Comment by chris-belcher** [comment]
The best privacy preseving method would be 4 where you import each UTXO private key into a newly generated wallet keeping the mixdepths the same.

Handling raw ECDSA keys is dangerous and unintuative. Option 3 would be best for safety.

**Comment by AdamISZ** [comment]
Yes, when I moved to a new wallet, I did one sweep per mixdepth. Perhaps this should be an "official" recommendation, not sure.
